tests/suite/math/call.typ

```diff
@@ -35,6 +126,34 @@ $ mat(#"code"; "wins") $
 #check($args(a,b;c)$, "(([a], [b]), ([c],))")
 #check($args(a,b;c,d;e,f)$, "(([a], [b]), ([c], [d]), ([e], [f]))")
 
+--- math-call-2d-named-repr ---
+#let args(..body) = (body.pos(), body.named())
+#let check(it, r) = test-repr(it.body.text, r)
+#check($args(a: b)$, "((), (a: [b]))")
+#check($args(1, 2; 3, 4)$, "((([1], [2]), ([3], [4])), (:))")
+#check($args(a: b, 1, 2; 3, 4)$, "((([1], [2]), ([3], [4])), (a: [b]))")
+#check($args(1, a: b, 2; 3, 4)$, "(([1], ([2],), ([3], [4])), (a: [b]))")
```

Comment:
Having 2 at a deeper array nesting level than 1, but the case it admittedly very cursed and I'm not sure whether we can improve it. 

Overall, for many of these mixed cases, it's anyone's guess on what the result is ...

Reply:
We probably can improve it I think. I kept it like this to match the current behaviour: it's a bit confusing but a named argument will update where an array can potentially start. In the example above, it was at `1`, then the named argument changed it to `2`, and so the following semicolon wraps the `2` in an array only.

Reply:
This is where runtime math parsing comes in :)

Reply:
Would that even been needed to do this? We could say store all the positional arguments as we loop and then at the semicolon build a SyntaxNode of an array with them manually.

Reply:
That would break the property that iterating over the syntaxnodes in order rebuilds the source text.

Reply:
This would be possible with changes at the AST layer though

Reply:
Ah, I see.

Reply:
How does runtime parsing fix it?

Reply:
With runtime parsing I can keep the dictionary and list of args completely separate, so if you do `$ func(1, 2; 3, named: x, 4) $` The named parameter doesn't interrupt the positional args being generated as `(1,2),(3,4)` in any way. We could also do this if we make the AST more aware of the 2-dimensional structure, but I am a bit biased.